src/coreclr/vm/jitinterface.h

```diff
@@ -439,6 +438,23 @@ class CEEInfo : public ICorJitInfo
     static size_t findNameOfToken (Module* module, mdToken metaTOK,
                             _Out_writes_ (FQNameCapacity) char * szFQName, size_t FQNameCapacity);
 
+#ifdef HOST_WINDOWS
+    static uint32_t ThreadLocalOffset(void* p);
+
+    // MethodTable to `typeIndex` map. `typeIndex` is embedded in the code during codegen. During execution
+    // corresponding thread static data blocks are stored in `t_threadStaticBlocks` array at the `typeIndex`.
+    static TypeIDMap g_threadStaticBlockTypeIDMap;
```

Comment:
TypeIDMap is designed for virtual stub dispatch. It generates "fat dispatch tokens" over certain limit. If you would like to reuse it here, I think it needs a different mode that suppresses the virtual stub dispatch specific features like fat dispatch tokens.

Reply:
Passing `bool useFatPointerDispatch`  parameter to `GetTypeID()` method. Let me know if that looks ok. I was considering passing a template parameter, but thought this approach was simpler.